homework/assignment_4/hw4.py

```diff
@@ -0,0 +1,216 @@
+# libraries
+import sys
+from io import StringIO
+
+import pandas as pd
+import plotly.express as px
+import pydot
+import statsmodels.api as sm
+from pandas import DataFrame
+from plotly import graph_objects as go
+from sklearn.model_selection import GridSearchCV
+from sklearn.tree import DecisionTreeClassifier, export_graphviz
+
+
+# function to for printing headings
+def print_heading(title):
+    print("*" * 80)
+    print(title)
+    print("*" * 80)
+    return
+
+
+# function to load any sklearn dataset in as pandas dataframe
+def sklearn_to_df(sklearn_dataset):
+    df = pd.DataFrame(sklearn_dataset.data, columns=sklearn_dataset.feature_names)
+    df["target"] = pd.Series(sklearn_dataset.target)
+    return df
+
+
+def plot_decision_tree(decision_tree, feature_names, class_names, file_out):
+    with StringIO() as dot_data:
+        export_graphviz(
+            decision_tree,
+            feature_names=feature_names,
+            class_names=class_names,
+            out_file=dot_data,
+            filled=True,
+        )
+        graph = pydot.graph_from_dot_data(dot_data.getvalue())
+        graph[0].write_pdf(file_out + ".pdf")  # must access graph's first element
+        graph[0].write_png(file_out + ".png")  # must access graph's first element
+
+
+# first pass testing on titanic
+def main():
+    df_titanic = pd.read_csv("./datasets/titanic.csv")
+    df_titanic = df_titanic.dropna()
+
+    # split for titanic
+    column_to_move = df_titanic.pop("Survived")  # remove survived - dependent variable
+    df_titanic[
+        "Survived"
+    ] = column_to_move  # add survived (DV) to end of df to split easier
+
+    # removing useless columns
+    df_titanic.pop("Ticket")
+    df_titanic.pop("Cabin")
+    df_titanic.pop("PassengerId")
+    df_titanic.pop("Name")
+
+    X = df_titanic.iloc[:, :-1]
+    y = df_titanic.iloc[:, -1]
+
```

Comment:
Im not sure how he'll input the columns and dataset but I think you may want to have clear areas of where you think the new dataset should go, probably right before new dataset is categorized by feature type.